### PR TITLE
Feature/dashboard calendar

### PR DESCRIPTION
**Pull request by jmerz826**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This pull request integrates a functioning calendar app (built with @paulstgermain ) so that an instructor may view his or her calendar, as well as create new events.  This a partial delivery because the events are not currently being pulled from and stored on the backend server, nor is the styling complete. This serves as a good jumping off point for the frontend team to customize the component to our UI team's vision!

Fixes # (issue)

## Loom Video
https://www.loom.com/share/044baf744dd14cb0a861124f20510b9d

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by ElchinNasirov — APPROVED** [review]

File: src/components/common/calendar/Calendar.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect } from 'react';
```
Comment (ryan-hamblin) [inline src/components/common/calendar/Calendar.js:1]: `useEffect` is defined but never used. Let's get rid of these warnings.

File: src/components/common/calendar/Calendar.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+// import './styles/App.css';
+import 'antd/dist/antd.css';
+import { Calendar, Modal, Badge, Button } from 'antd';
+import CalendarModal from './CalendarModal';
+import axiosWithAuth from '../../../utils/axiosWithAuth';
```
Comment (ryan-hamblin) [inline src/components/common/calendar/Calendar.js:6]: `axiosWithAuth` is defined but never used.

File: src/components/common/calendar/Calendar.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+function CalendarApp() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isScheduleModalVisible, setIsScheduleModalVisible] = useState(false);
+  const [event, setEvent] = useState(null);
+  const [eventsArr, setEventsArr] = useState(initialValues);
+
+  // ****** will update with endpoint once db endpoint is built ******
+  // useEffect(() => {
+  //     axiosWithAuth.get('/')
```
Comment (ryan-hamblin) [inline src/components/common/calendar/Calendar.js:61]: If we're keeping this as a stub, let's stub it out and uncomment. Commented-out code is against our standards.

File: src/components/common/calendar/Calendar.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  const showScheduleModal = () => {
+    setIsScheduleModalVisible(true);
+  };
+
+  function getListData(value, events) {
+    let listData = [];
+    let dateValue = value.format('DD/MM/YYYY'); // you can parse value in every format you want
+
+    events.map(e => {
+      if (e.date === dateValue) {
```
Comment (ryan-hamblin) [inline src/components/common/calendar/Calendar.js:90]: Not sure how much I care about this rule, but it's worth a refactor if ESLint is barking at it. You could simply substitute the `arrow func` for a `func declaration` ex. `function name()`

File: src/components/common/calendar/CalendarModal.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  };
+
+  const onFinish = values => {
+    const newEvent = {
+      ...values,
+      date: values.date.format('DD/MM/YYYY'),
+      time: values.time.format('h:mm A'),
+      type: 'success',
+    };
+    console.log('Success:', values);
```
Comment (ryan-hamblin) [inline src/components/common/calendar/CalendarModal.js:20]: Please remove this `console.log`

File: src/components/pages/InstructorHome/index.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import '../../../styles/InstructorStyles/index.less';
 import InstructorSidebar from './InstructorSidebar';
-import InstructorCalender from './InstructorCalender';
+// import InstructorCalender from './InstructorCalender';
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/index.js:4]: Let's not import this if we're not going to use it. Just remove the commented-out import.

File: src/components/pages/InstructorHome/index.js [hunk 7]
```diff
@@ -14,7 +15,8 @@ const InstructorHome = () => {
           <InstructorSidebar />
           <Banner />
           <div className="calendar" data-testid="calendar">
-            <InstructorCalender />
+            {/* <InstructorCalender /> */}
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/index.js:18]: Please remove commented out code.

**Review by ryan-hamblin — DISMISSED** [review]
Awesome implementation of the calendar! I'm eager to see others review this code as well. I think this is very close to being production ready. Just a few things to play around with. I'd also love to see it implemented and then have it backed by tests in the future.

**Review by jmerz826 — COMMENTED** [review]
Picked up and resolved @ryan-hamblin 's comments!

File: src/components/common/calendar/CalendarModal.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+export default function CalendarModal(props) {
+  const { isModalVisible, setIsModalVisible, eventsArr, setEventsArr } = props;
+
+  const [form] = Form.useForm();
+
+  const handleCancel = () => {
+    setIsModalVisible(false);
+  };
+
+  const onFinish = values => {
```
Comment (moua0061) [inline src/components/common/calendar/CalendarModal.js:13]: nice job on passing the props down

**Review by moua0061 — APPROVED** [review]

**Review by Raj-04 — APPROVED** [review]
Great work implementing the Calendar

**Pull request merged**